File: sol/706.py
class MyHashMap:
    CAPACITY = 1 << 10
    MASK = CAPACITY - 1

    def __init__(self):
        self.arr = tuple(deque() for _ in range(self.CAPACITY))

    def __computeHash(self, k: int) -> int:
        return ((k << 5) + k) & self.MASK

    def put(self, key: int, value: int) -> None:
        h = self.__computeHash(key)
        for i in range(len(self.arr[h])):
            if self.arr[h][i][0] == key:
                self.arr[h][i][1] = value
                return
        self.arr[h].append([key, value])

    def get(self, key: int) -> int:
        h = self.__computeHash(key)
        for i in range(len(self.arr[h])):
            if self.arr[h][i][0] == key:
                return self.arr[h][i][1]
        return -1

    def remove(self, key: int) -> None:
        h = self.__computeHash(key)
        for i in range(len(self.arr[h])):
            if self.arr[h][i][0] == key:
                self.arr[h][i][1] = -1
                return

File: sol/706/706.cpp
#include <deque>
#include <utility>

struct MyHashMap {
    constexpr static unsigned CAPACITY = 1 << 10;
    constexpr static unsigned MASK = (1 << 10) - 1;

    vector<deque<pair<int, int>>> arr;

    MyHashMap() {
        arr.resize(CAPACITY);
    }

    constexpr unsigned __computeHash(int key) const {
        return ((key << 5) + key) & MASK;
    }
    
    void put(int key, int value) {
        unsigned h = this->__computeHash(key);
        deque<pair<int, int>> &q = arr[h];
        for (deque<pair<int, int>>::iterator it = q.begin();
             it != q.end();
             ++it){
            if (it->first == key){
                it->second = value;
                return;
            }
        }
        q.emplace_back(make_pair(key, value));
    }
    
    int get(int key) const {
        unsigned h = this->__computeHash(key);
        const deque<pair<int, int>> &q = arr[h];
        for (deque<pair<int, int>>::const_iterator it = q.cbegin();
             it != q.cend();
             ++it){
            if (it->first == key){
                return it->second;
            }
        }
        return -1;
    }
    
    void remove(int key) {
        unsigned h = this->__computeHash(key);
        deque<pair<int, int>> &q = arr[h];
        for (deque<pair<int, int>>::const_iterator it = q.cbegin();
             it != q.cend();
             ++it){
            if (it->first == key){
                q.erase(it);
                return;
            }
        }
    }
};

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap* obj = new MyHashMap();
 * obj->put(key,value);
 * int param_2 = obj->get(key);
 * obj->remove(key);
 */
